ios/FluentUI/Navigation/BarButtonItemView.swift

```diff
@@ -0,0 +1,270 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+class BarButtonItemView: UIView {
+    let item: UIBarButtonItem
+    let isLeftItem: Bool
+
+    init(item: UIBarButtonItem, isLeftItem: Bool) {
+        self.item = item
+        self.isLeftItem = isLeftItem
+        super.init(frame: .zero)
+
+        container.addArrangedSubview(button)
+        container.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(container)
+
+        container.addSubview(badgeView)
+
+        isAccessibilityElement = true
+        updateAccessibilityLabel()
+
+        showsLargeContentViewer = true
+        scalesLargeContentImage = true
+
+        accessibilityHint = item.accessibilityHint
+        accessibilityIdentifier = item.accessibilityIdentifier
+
+        NSLayoutConstraint.activate([
```

Comment:
instead of this, you should be using intrinsiccontent size to define its width and height of this control